### PR TITLE
Warn if `SymbolPackageFormat` set to snupkg

### PR DESCRIPTION
**Pull request by RussKie**
To avoid the repeat of https://github.com/dotnet/extensions/issues/4406, warn if `SymbolPackageFormat` property is set to 'snupkg'.

If necessary, the warning can be suppressed with SuppressSymbolPackageFormatValidation=true.

. . .

Tested locally against https://github.com/dotnet/extensions/ @ ef72a4125a8f7f76b62024e273e90f53453b8f68.

With `<SymbolPackageFormat>snupkg</SymbolPackageFormat>`:
```
Build FAILED.

C:\Users\user\.nuget\packages\microsoft.dotnet.arcade.sdk\8.0.0-beta.23451.1\tools\Pack.targets(12,5): error : <SymbolPackageFormat>snupkg</SymbolPackageFormat> is not supported by the 
Arcade SDK. [D:\Development\dotnet-extensions\src\Libraries\Microsoft.AspNetCore.Testing\Microsoft.AspNetCore.Testing.csproj]
```

With the added suppression (`<SuppressSymbolPackageFormatValidation>true</SuppressSymbolPackageFormatValidation>`):
```
  Successfully created package 'D:\Development\dotnet-extensions\artifacts\packages\Debug\Shipping\Microsoft.AspNetCore.Testing.8.0.0-dev.nupkg'.
  Successfully created package 'D:\Development\dotnet-extensions\artifacts\packages\Debug\Shipping\Microsoft.AspNetCore.Testing.8.0.0-dev.snupkg'.
```

With the property removed:
```
  Successfully created package 'D:\Development\dotnet-extensions\artifacts\packages\Debug\Shipping\Microsoft.AspNetCore.Testing.8.0.0-dev.nupkg'.
  Successfully created package 'D:\Development\dotnet-extensions\artifacts\packages\Debug\Shipping\Microsoft.AspNetCore.Testing.8.0.0-dev.symbols.nupkg'.
```

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/sdk/Sdk.targets [hunk 1]
```diff
@@ -23,6 +23,8 @@
     </ItemGroup>
   </Target>
 
+  <Import Project="..\tools\Pack.targets" Condition="'$(__ImportPackTargets)' == 'true'" />
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/sdk/Sdk.targets:26]: I think this should actually be inside `Imports.targets`

**Review by mmitche — APPROVED** [review]

**Comment by RussKie** [comment]
@mmitche @dotnet/arcade-contrib may I please get some guidance with the failing tests? I don't think my change caused those as I see other PRs failing with the same error.

**Comment by ViktorHofer** [comment]
Just curious, what hinders Arcade from supporting snupkgs?

**Comment by RussKie** [comment]
> Just curious, what hinders Arcade from supporting snupkgs?

https://github.com/dotnet/arcade/issues/1959 <-- this?

**Comment by mmitche** [comment]
Nothing specific. The symbol publishing process needs to be updated though. You need to publish a binary with a symbol, and because snupkg doesn't contain the binary, this is a bit more complex than it is today.

But we should support snupkg

**Comment by hoyosjs** [comment]
The main problem with snupkg is that we can't easily convert portable PDBs to full PDBs without the PE. As the restriction goes away, snupkg makes sense and should be a straightforward migration.

**Pull request merged**